### ISSUE TITLE
a11y: Fix keyboard navigation, role, aria-label and focus display issues on the login screen

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12775]
a11y Issues: 49, 55, 43, 53

Fix keyboard navigation and focus issues on the logins screen:

![image](https://github.com/user-attachments/assets/467aea6d-32aa-4ade-ac7b-5fa0bd15ee59)

![image](https://github.com/user-attachments/assets/9159cfa3-c2c1-419a-aa39-12ce2fd44d56)

The locale menu should use the new Menu component.

**Issue 43**

The role and state is not defined for mentioned drop-down resulting in screen reader users are not able to understand the interactive elements functionality.

![image](https://github.com/user-attachments/assets/e4dc97e8-da69-4c6a-8ea6-07def9bc6b1a)

**Issue 53**

A focus indicator is not visually available as user tabs through mentioned checkbox using a keyboard resulting in users with learning and mobility impairments as well as those with low vision not being able to easily find out which element currently has the focus.

Login 
Main content
- Remember Username (unselected checkbox)

Homepage 
- Main Content 
- Logo image
- Cluster management" link
- Continuous Delivery
- Virtualization Management
- User and Authentication
and so on. 

Preferences
Main Content 
- Take me to the home page radio button
- Take me to the area I last visited radio button
- Take me to cluster: radio button 
- Enable "View in API"  checkbox 
and so on.

Account and API Keys
Main Content 
-  Checkboxes present inside the table
- Back button 

Create API Key
Main Content 
- 90  days - Maximum allowed radio button 
- Custom radio button 

Cluster Dashboard
Main Content 
- Cluster Tools 

Search
Main Content
- Admission
- MutatingWebhookConfigurations
- ValidatingAdmissionPolicies
- ValidatingAdmissionPolicyBindings
- ValidatingWebhookConfigurations 
and so on. 

Cluster Appearance
Main Content 
- Use custom badge, Show cluster comment, Badge background color checkboxes
- Transparent (available after selecting Badge background color) 

Projects/Namespaces
Main Content 
- Name, default, user-2rgrp and so on checkboxes

Create Namespace:
- Enforce
- Audit
- Warn

Charts
Main Content
- Show deprecated apps checkbox
- All, Rancher, Partners and RKE2 (that becomes available when user activate "All" dropdown list)

Charts: Kubecost (2.4.1)
Main content
- Show more / Show less

Cluster Management - Create
- RKE1 RKE2/3s switch

Cluster Drivers
- Alibaba ACK
- Amazon EKS
- Azure AKS
- Baidu CCE and so on

Pod Security Admission: Create
-Usernames, RuntimeClasses and Namespaces checkboxes 

Repository: Create
Main Content 
- http(s) URL to an index generated by Helm, Git repository containing Helm chart or cluster template definitions and OCI Repository radio buttons
- Remove buttons (which gets available after activating  Add Label and  Add Annotation button) 
- Skip TLS Verifications and Insecure Plain Http  (which gets available after activating OCI Repository Experimental radio buttons)

Git Repo - Add Repository
Main Content 
- Enable Self-Healing and Always Keep Resources checkboxes.
- Remove buttons (which gets available after activating  Add Path button) 

Users Retention Settings
Main Content 
- Disable user accounts after an inactivity period (duration since last login) and Delete user accounts after an inactivity period (duration since last login) checkboxes.
- Toggle button associated with Run the user retention process in DRY mode (no changes will be applied).

User: Create
Main Content 
- User, Show/hide and Release Notes links
- Ask user to change their password on next login, Generate a random password, Administrator, Restricted Administrator, Standard User and so on checkboxes. 

Add Extensions repositories
Main Content 
- Add Partners Extensions Repository checkbox. 

Global Settings
Main Content 
- Show Banner in Header, Show Banner in Footer, Show Consent Banner on Login Screen, Show Login Consent as a modal dialog, Show custom login error, Bold, Italic and Underline checkboxes 
- Left, Center and Right radio buttons. 
- Text Color and Background Color container                 
      
Branding
Main Content 
- Use a Custom Logo, Use a Custom Banner, Use a Custom Background, Use a Custom Favicon, Use a Custom Color and Use a Custom Link Color checkboxes.

Home Links
Main Content 
- Remove Button (available after activating Add Link button)
-  Show checkboxes 

About 
Main Content 
- Home Link 

![image](https://github.com/user-attachments/assets/a4efa9dd-d297-4416-b1b7-cca62492b89c)

^^ NOTE: The list goes beyond the login page - all need to be taking into account

**Comment by yonasberhe23** [comment]
Needs manual validation at this time as we dont have existing accessiblity tests. Follow-up ticket to create automated tests coverage https://github.com/rancher/qa-tasks/issues/1639

**Comment by IsaSih** [comment]
Reopening this issue, as tests fail in the Preferences page. When hit, the space bar doesn't just select the element but also scrolls the whole page down, as seen in the screenshots. The ideal behavior should only be pressing the space bar to select the element.

https://github.com/user-attachments/assets/4c3e604e-5a49-427b-a120-a464dc3dea34
https://github.com/user-attachments/assets/a4d997fd-9726-4123-9d0a-bcfb478ceaa3

**Comment by IsaSih** [comment]
Found bug in the Account and API keys page https://github.com/rancher/dashboard/issues/13237 that should be addressed separately, as it impacts keyboard navigation in dialogs from all pages

**Comment by IsaSih** [comment]
Found issue: https://github.com/rancher/dashboard/issues/13239 that needs to be addressed

**Comment by IsaSih** [comment]
Also found https://github.com/rancher/dashboard/issues/13240

**Comment by aalves08** [comment]
> Reopening this issue, as tests fail in the Preferences page. When hit inside dropdowns, the space bar doesn't just select the element but also scrolls the whole page down, as seen in the screenshots. The ideal behavior should only be pressing the space bar to select the element.
> 
> https://github.com/user-attachments/assets/cc26d287-dee7-487c-add8-422223986562 https://github.com/user-attachments/assets/6477da53-86b1-43ad-ad01-b014e2eb6b6b

@IsaSih , I've created https://github.com/rancher/dashboard/issues/13252 to track this.

With that, all of the items you mentioned are tracked on the accessibility board and will be triaged and prioritized.

Is any of them a release blocker or can we move this one forward and close it @nwmac @IsaSih ? FYI @rak-phillip

**Comment by IsaSih** [comment]
Also found https://github.com/rancher/dashboard/issues/13262

**Comment by IsaSih** [comment]
Question: also found this interesting behavior. Is this a bug?

https://github.com/user-attachments/assets/b03ab152-809b-4971-96f8-b86ed42cb8c6

**Comment by IsaSih** [comment]
I'm closing this issue as the bugs found during testing are being tracked separately.
